InnerEye/ML/model_testing.py

```diff
@@ -437,8 +440,27 @@ def test_epoch(test_epoch: int, run_recovery: Optional[RunRecovery]) -> Optional
         else:
             results[epoch] = epoch_result
 
+            if isinstance(epoch_result, ScalarMetricsDict):
+                epoch_folder = config.outputs_folder / get_epoch_results_path(epoch, data_split, model_proc)
+                csv_file = epoch_folder / METRICS_FILE_NAME
+
+                logging.info(f"Writing {data_split.value} metrics to file {str(csv_file)}")
+
+                # if we are running on the validation set, these may have been written during train time
+                if not csv_file.exists():
+                    os.makedirs(str(epoch_folder), exist_ok=False)
+                    df_logger = DataframeLogger(csv_file)
+
+                    # cross validation split index not relevant during test time
```

Comment:
How does this work if we are running a cross-validation run? One file per child run?

Reply:
When we have a set of cross validation runs, yes, this will create a csv per child run - this code doesn't support the cross validation case yet. On the other hand, if we run inference with checkpoints from a cross validation run, at that point we will have a single run and a single csv.